src/drivers/rc_epgsql.erl

```diff
@@ -0,0 +1,62 @@
+-module (rc_epgsql).
+
+-behaviour (gen_server).
+
+-export ([start_link/1]).
+-export ([init/1, handle_info/2, handle_cast/2, handle_call/3, terminate/2]).
+
+start_link(#{name := Name} = Args) ->
+  gen_server:start_link({local, Name}, ?MODULE, Args, []);
+
+start_link(Args) ->
+  gen_server:start_link({local, epgsql}, ?MODULE, Args, []).
+
+init(Args) ->
+  self() ! connect,
+  process_flag(trap_exit, true),
+  ValidArgs =
+    #{host         => maps:get(host, Args, "localhost"),
```

Comment:
how about something like
```erl
Defaults = #{host => "localhost", 
             username => os:getenv("USER"),
             password => "",
             database => "test_db",
             reconnection => {uniform, 4000}},
ValidArgs = maps:merge(Defaults, Args),
State = maps:merge(ValidArgs, #{retries => 0, state => no_connection})
```

Reply:
This solution could be possible but I found this approach harder to understand because you have to know that the `Args` map is going to prevail over `Defaults`map in `maps:merge(Defaults, Args)` also allow us to control the amount of settings that we will store in the gen_server State. Why do you think that is a better way to solve it?

Reply:
merge is kind of a common map operation, I think it's common knowledge that the second map overrides the first, I've seen this pattern commonly used to have default config values and allow to override them. Alos, I don't see how you lose control of the amount of settings in the state by doing this.

Reply:
You lose control of the amount of settings because `Args` could have any X parameter that the user adds in the configuration and they are going to be added in the `State`

Reply:
I don't think that's a big issue.

Reply:
@facundoolano is right that this is a common pattern, but for sane initialization and not for validation. The point is what you want to do here. If the point is to have a complete config with the good defaults overridden by what the user wants, the suggested code is appropriate. If the goal is to validate the user's config, the existing code is more appropriate.